src/sdk/BotFrameworkConfig.ts

```diff
@@ -56,16 +56,87 @@ export class BotFrameworkConfig extends DialogServiceConfigImpl {
      * @public
      * @param authorizationToken The authorization token associated with the bot
      * @param region The region name (see the <a href="https://aka.ms/csspeech/region">region page</a>).
+     * @param botId Optional. Identifier for using a specific bot within an Azure resource group. Equivalent to the resource name.
      * @returns {BotFrameworkConfig} A new bot framework config.
      */
-    public static fromAuthorizationToken(authorizationToken: string, region: string): BotFrameworkConfig {
+    public static fromAuthorizationToken(authorizationToken: string, region: string, botId?: string): BotFrameworkConfig {
         Contracts.throwIfNullOrWhitespace(authorizationToken, "authorizationToken");
         Contracts.throwIfNullOrWhitespace(region, "region");
 
         const botFrameworkConfig: BotFrameworkConfig = new DialogServiceConfigImpl();
-        botFrameworkConfig.setProperty(PropertyId.Conversation_DialogType, "bot_framework");
+        botFrameworkConfig.setProperty(PropertyId.Conversation_DialogType, DialogServiceConfig.DialogTypes.BotFramework);
         botFrameworkConfig.setProperty(PropertyId.SpeechServiceAuthorization_Token, authorizationToken);
         botFrameworkConfig.setProperty(PropertyId.SpeechServiceConnection_Region, region);
+
+        if (botId) {
+            botFrameworkConfig.setProperty(PropertyId.Conversation_ApplicationId, botId);
+        }
+
+        return botFrameworkConfig;
+    }
+
+    /**
+     * Creates an instance of a BotFrameworkConfig.
+     * This method is intended only for users who use a non-default service host. The standard resource path will be assumed.
+     * For services with a non-standard resource path or no path at all, use fromEndpoint instead.
+     * Note: Query parameters are not allowed in the host URI and must be set by other APIs.
+     * Note: To use an authorization token with fromHost, use fromHost(URL),
+     * and then set the AuthorizationToken property on the created BotFrameworkConfig instance.
+     * Note: Added in version 1.9.0.
+     * @member BotFrameworkConfig.fromHost
+     * @function
+     * @public
+     * @param {URL | string} host - If a URL is provided, the fully-qualified host with protocol (e.g. wss://your.host.com:1234)
+     *  will be used. If a string is provided, it will be embedded in wss://{host}.convai.speech.azure.us.
```

Comment:
Excellent explanation, this may save a GH issue or two.

Reply:
To be fair to everyone, it *is* awfully confusing.